### PR TITLE
Add option for custom colors

### PR DESCRIPTION
**Pull request by carzum**
Moving to structs for Subscriber configs, this will make custom configuration simpler.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -17,3 +16,11 @@ serde = { version = "*", features = ["derive"] }
 serde_derive = "*"
 termion = "1.5"
 tui = "0.10.0"
+
+[dependencies.confy]
+version = "0.4.0"
+features = ["yaml_conf"]
+default-features = false
+
+[patch.crates-io]
```
Comment (carzum) [inline Cargo.toml:25]: Confy > 0.5.0 is required as toml has bad handling of objects in lists. The maintainers 'forgot' to push to crates.io, thus the patch

File: src/config.rs [hunk 2]
```diff
@@ -19,15 +32,21 @@ impl Default for TermvizConfig {
...
-            target_framerate: 42,
+            map_topics: vec![ ListenerConfig {
+                topic: "map".to_string(),
+                color: Color { r: 255, b: 255, g: 255 } }],
+            laser_topics: vec![ListenerConfig {
+                topic: "scan".to_string(),
+                color: Color { r: 200, b: 0, g: 0 } }],
+            marker_array_topics: vec![ListenerConfig {
+                topic: "marker_array".to_string(),
+                color: Color { r: 20, b: 20, g: 200 } }],
```
Comment (carzum) [inline src/config.rs:43]: color for markers is not having an effect right now, its using the Colors from markers

File: src/config.rs [hunk 3]
```diff
@@ -19,15 +32,21 @@ impl Default for TermvizConfig {
...
+            target_framerate: 30,
             axis_length: 0.5,
-            visible_area: vec![5., -5., 5., -5.],
+            visible_area: vec![-5., 5., -5., 5.],
```
Comment (carzum) [inline src/config.rs:46]: The old default was just plain wrong. This took me waaaay to long to find

File: src/config.rs [hunk 4]
```diff
@@ -19,15 +32,21 @@ impl Default for TermvizConfig {
...
             zoom_factor: 0.1,
         };
-        let res = confy::store("termviz", &conf);
+        let res = confy::store("termviz", "termviz", &conf);
```
Comment (carzum) [inline src/config.rs:49]: Filenames can be given as args now

**Review by Eman7C7 — APPROVED** [review]

**Pull request merged**